rfc/sqlDesignDocs.md

```diff
@@ -0,0 +1,84 @@
+# Sql Design Docs
+
+## Overview
+
+Currently Fili only supports Druid for performing aggregations/grouping/post aggregations. Fili's processing pipeline is capable of sending queries to different backends, but lacks 
+
+1. the translation (serialization) from `DruidQuery` to other query formats. 
+2. the ability to manually build up a `ResultSet` and pass it through the `ResponseProcessor` instead of passing Druid's son responses
+
+This RFC aims to solve the first issue - lack of translation to another query model by using Calcite to build up SQL queries and processing the results into a Druid format ( `GroupByQuery` response) which Fili is able to understand.
+
+## Goals
+
+Allow customers to query Fili's API and make a `DruidAggregationQuery`  which will be processed by a SQL backed datasource instead of Druid.
+
+Customers will be able to configure their connection to a SQL database similar to Druid through the `applicationConfig.properties` and build a table in Fili specifically backed by SQL. They can then depend on the `fili`  and `fili-sql`module as opposed for configuring their application with SQL support.
+
+## Implementation
+
+1. Create a `fili-sql` module.
+
+2. Define a `SqlBackedClient` 
+
+   ```java
+   Future<JsonNode> executeQuery(
```

Comment:
The [`DruidWebService`](https://github.com/yahoo/fili/blob/2bd3ce3af7942ec485bb4bc84a37f7de21ea402a/fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/DruidWebService.java) got changed to return a future in #254 to clean up the blocking calls in `fili-generic-example`